### PR TITLE
Mssql docs

### LINKED ISSUE #713: add mssql to docs.
Hey 👋 

As MS SQL Server core dialect has been merged recently, we need to document it in preparation for the next release.

These places in the docs come to mind:

- "getting started".
- core dialects, wherever they're mentioned.

### PR DESCRIPTION
**Pull request by gittgott**
closes #713

I couldn't get deno or bun to work with the dialect. Deno was having issues connecting to the database with the message `Error: Failed to connect to localhost:1433 - tlssock._start is not a function`. Bun was having issues with tarn with the message `Failed to connect to localhost:1433 - socket must be an instance of net.Socket`.

Since this wasn't working for me, I stated that bun and deno didn't support the mssql dialect in the docs. Just want to make sure that they aren't currently known to work.

Thanks!

### CODE REVIEW

File: site/docs/getting-started/Dialects.tsx [hunk 1]
```diff
@@ -55,9 +60,10 @@ export function Dialects(props: DialectsProps) {
         it. This requires a <code>Dialect</code> implementation.
         <br />
         <br />
-        There are 3 built-in Node.js dialects for PostgreSQL, MySQL and SQLite.
-        Additionally, the community has implemented several dialects to choose
-        from. Find out more at <Link to="/docs/dialects">"Dialects"</Link>.
+        There are 4 built-in Node.js dialects for PostgreSQL, MySQL, SQLite, and
+        MSSQL. Additionally, the community has implemented several dialects to
+        choose from. Find out more at{' '}
```
Comment (igalklebanov) [inline site/docs/getting-started/Dialects.tsx:65]: - Let's use the official long name and abbreviation.
- Let's place it before SQLite as it is still more popular.

```suggestion
        There are 4 built-in dialects for PostgreSQL, MySQL, Microsoft SQL Server 
        (MSSQL), and SQLite. Additionally, the community has implemented several
        dialects to choose from. Find out more at{' '}
```

File: site/docs/getting-started/Summary.tsx [hunk 2]
```diff
@@ -37,6 +37,15 @@ const dialectSpecificCodeSnippets: Record<Dialect, string> = {
         cb.notNull().defaultTo(sql\`current_timestamp\`)
       )
       .execute()`,
+  mssql: `    await db.schema.createTable('person')
+      .addColumn('id', 'integer', (cb) => cb.primaryKey().modifyEnd(sql\`IDENTITY(1,1)\`))
```
Comment (igalklebanov) [inline site/docs/getting-started/Summary.tsx:41]: let's leave a TODO here to update this once we add `identity(1,1)` to core.

File: site/docs/getting-started/shared.tsx [hunk 3]
```diff
@@ -30,12 +30,24 @@ export const DRIVER_NPM_PACKAGE_NAMES: Record<Dialect, string> = {
...
+  mysql: undefined,
+  sqlite: undefined,
+  mssql: ['tarn'],
 }
 
 export const PRETTY_DIALECT_NAMES: Record<Dialect, string> = {
   postgresql: 'PostgreSQL',
   mysql: 'MySQL',
   sqlite: 'SQLite',
+  mssql: 'MSSQL',
```
Comment (igalklebanov) [inline site/docs/getting-started/shared.tsx:50]:
```suggestion
  mssql: 'Microsoft SQL Server (MSSQL)',
```

File: site/docs/getting-started/shared.tsx [hunk 4]
```diff
@@ -65,14 +77,19 @@ export interface Command {
...
   if (packageManager === 'deno') {
     throw new Error('Deno has no bash command')
   }
 
   return {
-    content: `${PACKAGE_MANAGER_INSTALL_COMMANDS[packageManager]} ${installedPackage}`,
+    content: `${
+      PACKAGE_MANAGER_INSTALL_COMMANDS[packageManager]
+    } ${installedPackage} ${
+      additionalPackages ? additionalPackages?.join(' ') : ''
```
Comment (igalklebanov) [inline site/docs/getting-started/shared.tsx:91]:
```suggestion
      additionalPackages ? additionalPackages.join(' ') : ''
```

**Review by igalklebanov — COMMENTED** [review]
Hey 👋 

Thanks! 💪 

Let some initial comments..

**Comment by igalklebanov** [comment]
We need to somehow mention that it also uses Tarn for connection pool management and provide a link to tarn's documentation.
https://github.com/kysely-org/kysely/pull/724/files#diff-f8ce7b5d9b471415c6e85f9db24baf58b16c5a55101d354bf66c0dc497340671R105-R109

**Comment by igalklebanov** [comment]
Regarding Deno compatibility, here's an official [issue](https://github.com/denoland/deno/issues/18303) that seems related to the problem (and someone claimed it still doesn't work 2 weeks ago despite being closed).

**Comment by igalklebanov** [comment]
Regarding Bun compatibility, here's an [issue](https://github.com/oven-sh/bun/issues/5147) that seems related to the problem (all commenters use libraries that use tedious under the hood).

**Comment by gittgott** [comment]
Hey @igalklebanov
I should have everything covered that you brought up.

Thanks for all your work maintaining this library by the way. You all do an amazing job.
I'm quite ecstatic that mssql is becoming part of the core dialects.

**Review by igalklebanov — COMMENTED** [review]
Looks good!

Just a small correction..

**Review by igalklebanov — CHANGES_REQUESTED** [review]
image
something's missing in "getting started -> querying".

**Comment by gittgott** [comment]
Hey @igalklebanov,

Sorry about that. I ran into some issues getting that one to work and forgot about it (and didn't run typecheck 🤦‍♂️)

I meant to ask: what would be a preferable solution for inserting a new row and returning data from that inserted row while #687 is still open?
The only thought I had was to use that existing findPeople function and grab the first of the array that's returned. Here's the snippet for that:
```typescript
export async function createPerson(person: NewPerson) {
  await db.insertInto('person')
    .values(person)
    .executeTakeFirstOrThrow()

  const newPerson = (await findPeople(person)).at(-1)
  return newPerson
}

export async function deletePerson(id: number) {
  const person = await findPersonById(id)

  if (person) {
    await db.deleteFrom('person').where('id', '=', id).execute()
  }

  return person
}
```

I think that the data will be sorted by id, so this would grab the most recently inserted row, but there would be issues if something was inserted to the table with identical values between the insertInto line and newPerson line were executed.

Am I missing something or would there be no better option until the output clause is supported?

**Comment by igalklebanov** [comment]
@gittgott I'd prefer using [scope identity](https://learn.microsoft.com/en-us/sql/t-sql/functions/scope-identity-transact-sql?view=sql-server-ver16&redirectedfrom=MSDN) in a transaction.

**Review by igalklebanov — APPROVED** [review]

**Pull request merged**